internal/provisioner/objects/dataplane/dataplane.go

```diff
@@ -421,7 +421,7 @@ func desiredDeployment(contour *model.Contour, contourImage, envoyImage string)
 						"prometheus.io/port":   "8002",
 						"prometheus.io/path":   "/stats/prometheus",
 					},
-					Labels: EnvoyPodSelector(contour).MatchLabels,
+					Labels: envoyPodLabels(contour),
```

Comment:
What's the reasoning for this change to add the component labels to the pod selector? The pod selector should already be unique per Gateway, not sure this is necessary.

Reply:
because when working with `helm`, it will auto add the labels(can't be removed)
```
  "app.kubernetes.io/name":       "xxx",
  "app.kubernetes.io/instance":   "yyy",
```
 to the service's `selector`, if not add the two labels to pod(s), the endpoint will be empty